improver/nbhood/square_kernel.py

```diff
@@ -111,7 +116,9 @@ def __init__(
         self.sum_only = sum_only
         self.re_mask = re_mask
 
-    def _calculate_neighbourhood(self, data: ndarray, mask: ndarray) -> ndarray:
+    def _calculate_neighbourhood(
+        self, data: ndarray, mask: ndarray = None
+    ) -> Union[ndarray, np.ma.MaskedArray]:
```

Comment:
As mentioned as something that could be considered for a separate PR, I think that potentially the following code block in this method can be simplified as shown below. This is because the "external mask" provided using the argument to the method is already incorporated into the full mask (combining the externally provided mask with a mask from the array) on line 148 and 151.

```
        area_mask_dtype = np.int64
        if mask is None:
            area_mask = np.ones(data.shape, dtype=area_mask_dtype)
        else:
            area_mask = np.array(mask, dtype=area_mask_dtype, copy=False)
```

to

```
area_mask = np.ones(data.shape, dtype=np.int64)
```

Reply:
I'll do this in a follow up PR